src/utmpx.rs

```diff
@@ -181,20 +192,34 @@ impl Utmp {
         unsafe { libc::setutxent() }
     }
 
-    #[cfg(target_env = "gnu")]
     /// Set default path (GNU extension).
-    fn set_default_path() -> Result<CString> {
-        let path = CString::new(UTMPX_FILENAME).map_err(|_| Error::InvalidPath)?;
+    #[cfg(target_env = "gnu")]
+    fn set_default_path() -> Result<()> {
+        use std::ffi::CString;
+
+        let path = CString::new(UTMPX_DEFAULT_PATH).map_err(|_| Error::InvalidPath)?;
```

Comment:
I've left a TODO here, I'll followup on this. It is a string only used internally, if possible I'd rather not block this on future libc API.

Reply:
I've added a TODO comment for this, I'm gradually following up on libc. If possible, I'd rather not block this on future libc API.

Reply:
No, I really don't want to define any magic strings in Nix.  That string is part of the `utmpx` internal implementation, not its public API.  It will only cause problems to define it here.